### PR TITLE
Add test for contextlib module

### PR DESCRIPTION
**Pull request by m4ns0ur**

### CODE REVIEW

File: third_party/stdlib/test/test_contextlib.py [hunk 1]
```diff
@@ -0,0 +1,353 @@
...
+            state.append(1)
+            yield 42
+            state.append(999)
+        with woohoo() as x:
+            self.assertEqual(state, [1])
+            self.assertEqual(x, 42)
+            state.append(x)
+        self.assertEqual(state, [1, 42, 999])
+
+    @unittest.skip('grumpy')
```
Comment (trotterdylan) [inline third_party/stdlib/test/test_contextlib.py]: ISTM that most of these tests should work. Is there something specific that's broken preventing them from working?

Comment (m4ns0ur) [inline third_party/stdlib/test/test_contextlib.py]: Yes, 'stat._getframe()' is missing.
not sure how deap it's gonna be, but I can give it a shot.

Comment (trotterdylan) [inline third_party/stdlib/test/test_contextlib.py]: Note that Grumpy has a built in called \_\_frame\_\_() that gets the top level fame and you can use f_back to get subsequent ones.

**Comment by m4ns0ur** [comment]
@trotterdylan it looks better now, PTAL.

BTW couple of test cases are failing, because raising an error in `with - generator` block, is not triggering `generator - finally`, any thought why?

File: runtime/dict_test.go [hunk 2]
```diff
@@ -615,6 +615,19 @@ func TestDictNewRaises(t *testing.T) {
 	}
 }
 
+func TestDictSetDefault(t *testing.T) {
+	cases := []invokeTestCase{
+		{args: wrapArgs(newTestDict("foo", 42), "foo"), want: NewInt(42).ToObject()},
```
Comment (trotterdylan) [inline runtime/dict_test.go]: You should also check that the dictionary is updated appropriately. One way to do that is to define a function that returns the result of setdefault and the dictionary itself and comparing against that. See TestDictSetItem for an example.

**Comment by m4ns0ur** [comment]
it's too far from master already, so I'm closing this one. will create new one after `generator.throw()` PR.

**Pull request closed without merging**